common/components/AddressBookTable.tsx

```diff
@@ -0,0 +1,178 @@
+import KeyCodes from 'shared/keycodes';
+import { AppState } from 'reducers';
+import translate from 'translations';
+import { isValidETHAddress } from 'libs/validators';
+import {
+  TAddLabelForAddress,
+  TRemoveLabelForAddress,
+  addLabelForAddress,
+  removeLabelForAddress,
+  AddressLabelPair
+} from 'actions/addressBook';
+import { getAddressLabelPairs } from 'selectors/addressBook';
+import { Input, Identicon } from 'components/ui';
+import AddressBookTableRow from './AddressBookTableRow';
+import './AddressBookTable.scss';
+
+interface DispatchProps {
+  addLabelForAddress: TAddLabelForAddress;
+  removeLabelForAddress: TRemoveLabelForAddress;
+}
+
+interface StateProps {
+  rows: ReturnType<typeof getAddressLabelPairs>;
+}
+
+type Props = DispatchProps & StateProps;
+
+interface State {
+  editingRow: number | null;
+  temporaryLabel: string;
+  temporaryAddress: string;
+}
+
+class AddressBookTable extends React.Component<Props, State> {
+  public state: State = {
+    editingRow: null,
+    temporaryLabel: '',
+    temporaryAddress: ''
+  };
+
+  private addressInput: HTMLInputElement | null = null;
+
+  private labelInput: HTMLInputElement | null = null;
+
+  public render() {
+    const { rows } = this.props;
+    const { temporaryLabel, temporaryAddress } = this.state;
+
+    return (
+      <table className="AddressBookTable table" onKeyDown={this.handleKeyDown}>
+        <thead>
+          <tr>
+            <th scope="col">{translate('ADDRESS')}</th>
+            <th scope="col">{translate('LABEL')}</th>
+            <th scope="col">{translate('ACTIONS')}</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              <div className="AddressBookTable-cell">
+                <Input
+                  placeholder="New address"
+                  value={temporaryAddress}
+                  onChange={this.setTemporaryAddress}
+                  setInnerRef={this.setAddressInputRef}
+                />
+                <Identicon address={temporaryAddress} />
+              </div>
+            </td>
+            <td>
+              <div className="AddressBookTable-cell">
+                <Input
+                  placeholder="New label"
+                  value={temporaryLabel}
+                  onChange={this.setTemporaryLabel}
+                  setInnerRef={this.setLabelInputRef}
+                />
+              </div>
+            </td>
+            <td>
+              <div className="AddressBookTable-cell">
+                <button className="btn btn-sm btn-default" onClick={this.handleAddEntry}>
+                  <i className="fa fa-plus" />
+                </button>
+              </div>
+            </td>
+          </tr>
+          {rows.map(this.makeLabelRow)}
+        </tbody>
+      </table>
+    );
+  }
+
+  private handleSave = (addressToLabel: AddressLabelPair) => {
+    const { label, address } = addressToLabel;
+
+    this.props.addLabelForAddress({ label, address });
```

Comment:
No need to destructure here, just pass the arg along

Reply:
👍